.github/workflows/docker-build.yaml

```diff
@@ -27,13 +27,43 @@ jobs:
         shell: bash
         run: |
           cd tests/basic
-          for f in *.py; do echo "$f" && python "$f"; done
+          for f in *.py; do echo "$f" && python "$f" &>> basic.log; done
+          done=$(cat basic.log | grep -c "DONE")
+          if [[ $done == 4 ]]
+          then
+            echo "OK"
+            exit 0
+          else
+            echo "not OK"
```

Comment:
nit: could we make the failure message a bit more explicit? 
maybe something like: `FAILED: One or more programs failed during execution`